### PR TITLE
Convert ResourceMapper extract and populate functions to suspend funcs

### LINKED ISSUE #666: Convert ResourceMapper extraction function to suspend function
- [x] Convert ResourceMapper extraction and population function to suspend function
- [x] Convert NpmPackageProvider & SimpleWorkerContextProvider functions to suspend functions

### PR DESCRIPTION
**Pull request by ekigamba**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #666 

**Description**
Convert ResourceMapper, SimpleWorkerContextProvider and NpmPackageProvider functions to suspend functions

**Alternative(s) considered**
None

**Type**
Feature

**Screenshots (if applicable)**
None

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 1]
```diff
@@ -144,15 +144,20 @@ object ResourceMapper {
 
     return Bundle().apply {
       StructureMapUtilities(simpleWorkerContext)
-        .transform(simpleWorkerContext, questionnaireResponse, structureMap, this)
+        .transform(
+          /* appInfo= */ simpleWorkerContext,
+          /* source= */ questionnaireResponse,
+          /* map= */ structureMap,
+          /* target= */ this
+        )
```
Comment (aditya-07) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:152]: nit: Are the named arguments required here. Can they be removed?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 2]
```diff
@@ -101,7 +101,7 @@ object ResourceMapper {
    * extracted resource. If the process completely fails, an error is thrown or a [Bundle]
    * containing empty [Resource] is returned
    */
-  private fun extractByDefinitions(
+  private suspend fun extractByDefinitions(
```
Comment (aditya-07) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:104]: All the heavy duty private functions in the file called by `extractByDefinitions ` should also be marked `suspend` e.g. `extractFields`, `extractField`

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 3]
```diff
@@ -144,15 +144,20 @@ object ResourceMapper {
...
+        )
     }
   }
 
   /**
    * Returns a `QuestionnaireResponse` to the [questionnaire] that is pre-filled from the [resource]
    * See http://build.fhir.org/ig/HL7/sdc/populate.html#expression-based-population.
    */
-  fun populate(questionnaire: Questionnaire, resource: Resource): QuestionnaireResponse {
+  suspend fun populate(questionnaire: Questionnaire, resource: Resource): QuestionnaireResponse {
```
Comment (aditya-07) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:160]: `populateInitialValues`, `populateInitialValue` should also be suspending functions.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/utilities/NpmPackageProvider.kt [hunk 4]
```diff
@@ -59,7 +59,7 @@ object NpmPackageProvider {
    *
    * The whole process can take 1-2 minutes on a clean installation.
    */
-  fun loadNpmPackage(context: Context): NpmPackage {
+  suspend fun loadNpmPackage(context: Context): NpmPackage {
```
Comment (aditya-07) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/utilities/NpmPackageProvider.kt:62]: `setupNpmPackage` should be a marked suspend.

**Review by aditya-07 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt (hunk 1 above)
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:152]: Cool

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt (hunk 2 above)
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:104]: Done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt (hunk 3 above)
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:160]: Done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/utilities/NpmPackageProvider.kt (hunk 4 above)
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/utilities/NpmPackageProvider.kt:62]: Done

**Review by aditya-07 — APPROVED** [review]
LGTM

**Pull request merged**